cellarium/ml/models/scvi.py

```diff
@@ -0,0 +1,521 @@
+            ``encode_covariates`` is ``True``.
+        use_batch_norm:
+            Specifies where to use :class:`~torch.nn.BatchNorm1d` in the model. One of the following:
+                * ``"none"``: don't use batch norm in either encoder(s) or decoder.
+                * ``"encoder"``: use batch norm only in the encoder(s).
+                * ``"decoder"``: use batch norm only in the decoder.
+                * ``"both"``: use batch norm in both encoder(s) and decoder.
+            Note: if ``use_layer_norm`` is also specified, both will be applied (first
+            :class:`~torch.nn.BatchNorm1d`, then :class:`~torch.nn.LayerNorm`).
+        use_layer_norm:
+            Specifies where to use :class:`~torch.nn.LayerNorm` in the model. One of the following:
+                * ``"none"``: don't use layer norm in either encoder(s) or decoder.
+                * ``"encoder"``: use layer norm only in the encoder(s).
+                * ``"decoder"``: use layer norm only in the decoder.
+                * ``"both"``: use layer norm in both encoder(s) and decoder.
+            Note: if ``use_batch_norm`` is also specified, both will be applied (first
+            :class:`~torch.nn.BatchNorm1d`, then :class:`~torch.nn.LayerNorm`).
+        use_size_factor_key:
+            If ``True``, use the :attr:`~anndata.AnnData.obs` column as defined by the
+            ``size_factor_key`` parameter in the model's ``setup_anndata`` method as the scaling
+            factor in the mean of the conditional distribution. Takes priority over
+            ``use_observed_lib_size``.
+        use_observed_lib_size:
+            If ``True``, use the observed library size for RNA as the scaling factor in the mean of the
+            conditional distribution.
+        library_log_means:
+            :class:`~numpy.ndarray` of shape ``(1, n_batch)`` of means of the log library sizes that
+            parameterize the prior on library size if ``use_size_factor_key`` is ``False`` and
+            ``use_observed_lib_size`` is ``False``.
+        library_log_vars:
+            :class:`~numpy.ndarray` of shape ``(1, n_batch)`` of variances of the log library sizes
+            that parameterize the prior on library size if ``use_size_factor_key`` is ``False`` and
+            ``use_observed_lib_size`` is ``False``.
+        var_activation:
+            Callable used to ensure positivity of the variance of the variational distribution. Passed
+            into :class:`~scvi.nn.Encoder`. Defaults to :func:`~torch.exp`.
+        extra_encoder_kwargs:
+            Additional keyword arguments passed into :class:`~scvi.nn.Encoder`.
+        extra_decoder_kwargs:
+            Additional keyword arguments passed into :class:`~scvi.nn.DecoderSCVI`.
+        batch_embedding_kwargs:
+            Keyword arguments passed into :class:`~scvi.nn.Embedding` if ``batch_representation`` is
+            set to ``"embedding"``.
+    """
+
+    def __init__(
```

Comment:
Additional flexibility that is needed:

- Which layer of the encoder and/or decoder to include batch variables; "-1" would mean nowhere!
- The option to either learn batch variables, to use "batch vectors" that are optionally provided to the initializer